src/contracts/AccessControlShares.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {MultiLP} from "./MultiLP.sol";
+
+abstract contract AccessControlShares is MultiLP {
+    uint256 public totalSupplyCap;
+    mapping(address lp => uint256 shares) public liquidityProviderCaps;
+
+    uint256[50] private _gap;
+
+    event LiquidityProviderCap(address indexed liquidityProvider, uint256 cap);
+    event TotalSupplyCap(uint256 shares);
+
+    function _postDepositHook(uint256) internal virtual override {
```

Comment:
I know we are moving this logic elsewhere, but we should also:
- decrement the individual cap
- emit the same set event with the new individual cap after the decrement